bin/main.ml

```diff
@@ -13,23 +13,25 @@ let rec is_duplicate = function
 | h1::h2::t -> if h1 = h2 then is_duplicate (h2::t) else false
 | _ -> true
 
-(* this is quite slow *)
-let rec is_unique = function
-| [] -> true
-| h::t -> if List.mem h t then false else is_unique t
+let is_unique l = 
+  let len = List.length l in
+  let set = Hashtbl.create len in
+  let () = List.iter (fun x -> Hashtbl.replace set x ()) l in 
+  len = Hashtbl.length set
 
-let precondition_frequency prop name =
+let default_gen = Combinators.int_list_sorted
+let precondition_frequency prop name gen_type =
   QCheck.(Test.make
   ~count:20000
   ~name
-   (pair (int) (list int)) (fun (n, l) ->
+   (pair (int) (gen_type)) (fun (n, l) ->
```

Comment:
Not all functions will need this pair of int/gen_type. We can instead replace this with just gen_type and have each benchmark fully supply it's required generator